src/machine/machinectl.c

```diff
@@ -186,16 +245,20 @@ static int list_machines(int argc, char *argv[], void *userdata) {
         qsort_safe(machines, n_machines, sizeof(MachineInfo), compare_machine_info);
 
         if (arg_legend && n_machines > 0)
-                printf("%-*s %-*s %-*s\n",
+                printf("%-*s %-*s %-*s %-*s\n",
                        (int) max_name, "MACHINE",
                        (int) max_class, "CLASS",
-                       (int) max_service, "SERVICE");
+                       (int) max_service, "SERVICE",
+                       (int) max_os, "OS");
 
-        for (j = 0; j < n_machines; j++)
-                printf("%-*s %-*s %-*s\n",
+        for (j = 0; j < n_machines; j++) {
+                printf("%-*s %-*s %-*s %-*s\n",
                        (int) max_name, machines[j].name,
                        (int) max_class, machines[j].class,
-                       (int) max_service, machines[j].service);
+                       (int) max_service, machines[j].service,
+                       (int) max_os, machines[j].os ? machines[j].os : "");
+                free(machines[j].os);
```

Comment:
Hm, this cleanup is not very pretty, and the memory will be leaked in error paths. Fixing it would be complicated, but I think a better solution is to not store the name in machines array, but just query it inside of this loop. Since the last column is left aligned, the width of the last column is not used for anything. So you can just call `get_os_name` right here.